include/openthread/platform/infra_if.h

```diff
@@ -67,6 +67,17 @@ extern "C" {
  */
 bool otPlatInfraIfHasAddress(uint32_t aInfraIfIndex, const otIp6Address *aAddress);
 
+/**
+ * Tells whether a given IPv6 address has on-link prefix for a given infrastructure interface index.
+ *
+ * @param[in]  aInfraIfIndex  The index of the infrastructure interface.
```

Comment:
How about we also mention that the input `aAddress` may be an IPv4 address which is then represented as an IPv4-mapped IPv6 address?